lib/Net/Google/Drive/Simple/V3.pm

```diff
@@ -661,37 +667,113 @@ sub upload_file {
             'useContentAsIndexableText' => [ TYPE_BOOLEAN(), 0 ]
         },
 
-        'body_parameters' => [ qw<
-            appProperties
-            contentHints
-            contentRestrictions
-            copyRequiresWriterPermission
-            createdTime
-            description
-            folderColorRgb
-            id
-            mimeType
-            modifiedTime
-            name
-            originalFilename
-            parents
-            properties
-            shortcutDetails.targetId
-            starred
-            viewedByMeTime
-            writersCanShare
-        > ],
+        'path'                => $self->{'api_upload_url'},
+        'method_name'         => 'upload_media_file',
+        'http_method'         => HTTP_METHOD_POST(),
+        'body_content'        => $self->_content_sub($file),
+        'extra_headers'       => [
+            'Content-Type'   => $mimeType,
+            'Content-Length' => $size,
+        ],
+    };
 
-        'path'        => $self->{'api_upload_url'},
-        'method_name' => 'upload_file',
-        'http_method' => HTTP_METHOD_POST(),
+    if ( defined $options->{'enforceSingleParent'} ) {
+        LOGDIE("[$info->{'method_name'}] Creating files in multiple folders is no longer supported");
+    }
+
+    # Since a file upload can take a long time, refresh the token
+    # just in case.
+    $self->{'oauth'}->token_expire();
+
+    return $self->_handle_api_method( $info, $options );
+}
+
+###########################################
+sub upload_multipart_file {
+###########################################
+    my ( $self, $file, $options ) = @_;
+
+    defined $file && length $file
+        or LOGDIE('upload_multipart_file() missing file');
+
+    -r $file
+        or LOGDIE("upload_multipart_file() received non-existent/unreadable file: $file");
+
+    my $size = -s $file;
+    $size <= 5_242_880 # 5M
```

Comment:
maybe good usage of a constant there? we seem to reuse that limit

Reply:
Good call. Done.